exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/AsyncPageReader.java

```diff
@@ -409,9 +414,10 @@ public Void call() throws IOException {
 
       DrillBuf pageData = null;
       timer.reset();
-      try {
+      parqProcessing: try {
```

Comment:
Maybe just check isShuttingDown here and return? (breaks with labels with no outer for/while loops confuse me!)